From 701fbb54644d27e754f9c15397adcd761de57f4e Mon Sep 17 00:00:00 2001
From: Sergey Shelukhin <sershe@apache.org>
Date: Tue, 12 Jul 2016 21:24:01 -0700
Subject: [PATCH 1044/1164] CDH-51807: HIVE-14210 : ExecDriver should call
 jobclient.close() to trigger cleanup (Thomas
 Friedrich, reviewed by Sergey Shelukhin and
 Vaibhav Gumashta)

(cherry picked from commit 1884d272be4deb0b75eedd0e441096ea79af4419)

Change-Id: I0a9934e83b50c24a4df1ec7e65c50c560541bfcc
---
 .../apache/hadoop/hive/ql/exec/mr/ExecDriver.java  |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/ql/src/java/org/apache/hadoop/hive/ql/exec/mr/ExecDriver.java b/ql/src/java/org/apache/hadoop/hive/ql/exec/mr/ExecDriver.java
index e09ac8d..b9319cb 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/exec/mr/ExecDriver.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/exec/mr/ExecDriver.java
@@ -222,6 +222,7 @@ public int execute(DriverContext driverContext) {
     Context ctx = driverContext.getCtx();
     boolean ctxCreated = false;
     Path emptyScratchDir;
+    JobClient jc = null;
 
     if (driverContext.isShutdown()) {
       LOG.warn("Task was cancelled");
@@ -389,7 +390,7 @@ public int execute(DriverContext driverContext) {
       if (pwd != null) {
         HiveConf.setVar(job, HiveConf.ConfVars.METASTOREPWD, "HIVE");
       }
-      JobClient jc = new JobClient(job);
+      jc = new JobClient(job);
       // make this client wait if job tracker is not behaving well.
       Throttle.checkJobTracker(job, LOG);
 
@@ -472,6 +473,9 @@ public int execute(DriverContext driverContext) {
           HadoopJobExecHelper.runningJobs.remove(rj);
           jobID = rj.getID().toString();
         }
+        if (jc!=null) {
+          jc.close();
+        }
       } catch (Exception e) {
       }
     }
-- 
1.7.9.5

